src/components/TransactionResult.tsx

```diff
@@ -0,0 +1,53 @@
+import {
+  Alert,
+  AlertDescription,
+  AlertIcon,
+  AlertTitle,
+  Flex,
+} from "@chakra-ui/react";
+
+export enum ActionName {
+  Deposit = "Deposit",
+  Withdraw = "Withdraw",
+  Approve = "Approve",
+  Associate = "Associate",
+  Mint = "Mint",
+  Claim = "Claim",
+}
+
+const getAlertTitle = (isSuccess = true, actionType?: ActionName) => {
+  return `${actionType} ${isSuccess ? "success" : "error"}`;
+};
+
+export const TransactionResult = ({
+  actionName,
+  transactionResult,
+  transactionError,
+}: {
+  actionName: ActionName | undefined;
```

Comment:
What is the case when `actionName` is undefined?

I'd also extend the type to allow passing just plain strings, would make it more flexible